internal/driver/deviceclient.go

```diff
@@ -39,13 +39,16 @@ type CommandInfo struct {
 	RawType    sdkModel.ValueType
 }
 
-func createCommandInfo(req *sdkModel.CommandRequest) (*CommandInfo, error) {
+//createCommandInfo version to handle zero base addressing
+func createCommandInfo(req *sdkModel.CommandRequest, zeroBase bool) (*CommandInfo, error) {
 	primaryTable, _ := req.Attributes[PRIMARY_TABLE]
 	primaryTable = strings.ToUpper(primaryTable)
 
 	startingAddress, _ := toUint16(req.Attributes[STARTING_ADDRESS])
-	startingAddress = startingAddress - 1
-
+	//startingAddress = startingAddress - 1
+	if !zeroBase {
```

Comment:
Should be `if zeroBase {}` As you mentioned:
> You could always add a bool value that represents “zero based addressing”. I’ve seen this in many modbus implementations. If true then -1 else do nothing.

https://github.com/edgexfoundry/device-modbus-go/issues/61#issuecomment-674454763